src/components/achievements.tsx

```diff
@@ -0,0 +1,127 @@
+'use client';
+
+import React from 'react';
+import { Doughnut } from 'react-chartjs-2';
+import {
+  Chart as ChartJS,
+  ChartData,
+  Tooltip,
+  ArcElement,
+  Filler,
+  Legend,
+  Title,
+} from 'chart.js/auto';
+
+ChartJS.register(ArcElement, Tooltip, Legend, Title, Filler);
+
+interface AchievementData {
+  difficulty: number; // 難易度 (1: Easy, 2: Normal, 3: Hard)
+  status: 'submitted' | 'reviewed' | '';
+}
```

Comment:
`@/lib/submissions` で型を定義されているので、
そちらを使いましょう！
```suggestion
import { Submission } from '@/lib/submissions';

interface AchievementData {
  difficulty: Submission['difficulty'];
  status: Submission['status'];
}
```

Reply:
現段階では OK ですが、
このあたりは、今後 Strapi にアクセスする API を作る際に確定すると思うので、
その際にはそちらを採用しましょう！
その際にはこのようになると思います。
```typescript
import { Difficulty } from '@/domain/entities/problem';
import { Status } from '@/domain/entities/submission';

interface AchievementData {
  difficulty: Difficulty;
  status: Status;
}
```

Reply:
[6300fd6](https://github.com/Greek-Academy/alpha-dojo-v2/pull/43/commits/6300fd680607ed0717ee172f5e492e0142bfb4eb)で対応済み